src/middlewared/middlewared/plugins/snapshot_/removal_date.py

```diff
@@ -0,0 +1,23 @@
+import hashlib
+
+from middlewared.service import Service, job, private
+
+
+class PeriodicSnapshotTaskService(Service):
+
+    class Config:
+        namespace = "pool.snapshottask"
+
+    @private
+    async def removal_date_property(self):
```

Comment:
Probably should make this synchronous since it can block the event loop.

Reply:
Also, can you use the `system.host_id` method which basically does the same thing?